packages/create-app/index.js

```diff
@@ -122,13 +122,7 @@ async function init() {
 
   const pkg = require(path.join(templateDir, `package.json`))
 
-  pkg.name = path
-    .basename(root)
-    // #2360 ensure packgae.json name is valid
-    .trim()
-    .replace(/\s+/g, '-')
-    .replace(/^[._]/, '')
-    .replace(/[~)('!*]+/g, '-')
+  pkg.name = packageName
```

Comment:
**suggestion:** We now may call `pkg.name = await getValidPackageName(targetDir)` here, so the local variable is not so much far away from it's usage